File: server/events/classes/fire_ack_event.py
from .event import Event

class FireAckEvent(Event):

    def __init__(self, toUnit, Id:int, gameId:int, victimDead:bool, allDeadGiveUp:bool, miss:bool, dodge:bool, criticalHit:bool):
        self["eventType"] = "fire-ack"
        self["toUnit"] = toUnit
        self["id"] = Id
        self["gameId"] = gameId
        self["victimDead"] = victimDead
        self["allDeadGiveUp"] = allDeadGiveUp
        self["miss"] = miss
        self["dodge"] = dodge
        self["criticalHit"] =criticalHit

File: vis-viva/app/src/model/events/classes/FireAckEvent.js
import BattleUnit from "../../battle-units/BattleUnit";
import BattleUnitFactory from "../../battle-units/BattleUnitFactory";
import RemoteEvents from "../RemoteEvents";
import Event from "../Event.js"


export default class FireAckEvent extends Event{

    /**
     * 
     * @param {{
     * gameId: number,
     * toUnit: {},
     * id: number,
     * victimDead: boolean,
     * allDeadGiveUp: boolean,
     * miss : boolean,
     * dodge : boolean,
     * criticalHit :boolean
     * }} json 
     */
    constructor(json){
        super(RemoteEvents.FIRE_ACK)

        this.gameId = json.gameId
        this.toUnit = BattleUnitFactory.fromJson(json.toUnit)
        this.id = json.id
        this.victimDead = json.victimDead
        this.allDeadGiveUp = json.allDeadGiveUp
        this.miss = json.miss
        this.dodge = json.dodge
        this.criticalHit = json.criticalHit
    }
    
}



